src/transformStateWithClones.js

```diff
@@ -7,7 +7,59 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesHistory = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    stateCopy = { ...stateCopy };
```

Comment:
This line is redundant as you are already creating a copy of the state at the beginning of the function (line 11). The stateCopy will be a new object in each iteration of the loop, so you don't need to copy it again here.